BlazorApp1/wwwroot/blazor-fusionCharts.js

```diff
@@ -1,33 +1,72 @@
-            let regEx = new RegExp("{([^}]*)}");
-            let func = new Function(value.split("function()")[1]);
-            return func;
-        }
-        return value
-    }
-    // Responsible for rendering any chart//
-    window.FusionCharts.renderChart = (chartConfiguration) => {
-        options = JSON.parse(chartConfiguration, this.parseFunction);
-        const chart = new FusionCharts(options);
-        chart.render();
-    };
-    //Generic Method to call any fusion chart method exluding except above methods//
-    window.FusionCharts.invokeChartFunction = (functionName, chartID, ...args) => {
-        var currentChart = FusionCharts(chartID);
-        var result =   currentChart[functionName].apply(currentChart, ...args);
-        var typeofresult = typeof result;
-        if(typeofresult === "number" || typeofresult === "boolean"){
-            result = result.toString();
-        }
-        else if(typeofresult === "object"){
-            result = JSON.stringify(result);
-        }
-        else if(typeofresult === "XML"){
-            result = XML.stringify(result);
-        }
-        if(typeofresult === "string"){
-            return result;
-        }
-        return result.toString() || "";
-    };
\ No newline at end of file
+  if (typeof value === "string" && value.includes(FUNCTION_KEYWORD)) {
+    let functionParts = value.split("function()")
+    let functionBody = functionParts[1]
+    let parsedFn = new Function(functionBody);
+    return parsedFn;
+  }
+  // if the value is not function then return as is
+  return value;
+}
+// Responsible for rendering any chart//
+window.FusionCharts.renderChart = (chartConfiguration) => {
+  const configAsJSObject = JSON.parse(chartConfiguration, this.parseFunction);
+  const chart = new FusionCharts(configAsJSObject);
+  chart.render();
+};
+
+// Method written for charts like time-series chart
+window.FusionCharts.setDataStore = (id, args) =>  {
+
+    var currentChart = FusionCharts(id);
+    var fusionDataStore = new FusionCharts.DataStore();
+    
+    var data = JSON.parse(args[0]);
+    var schema = JSON.parse(args[1]);
+
+    var fusionTable = fusionDataStore.createDataTable(data, schema);
+
+    currentChart.setChartData({data: fusionTable});
+};
+
+// resizeTo will not return any data due to circular json object
+window.FusionCharts.resizeTo = (id, args) => {
+    var currentChart = FusionCharts(id);
+    currentChart.resizeTo(args[0], args[1]);
+}
+
+//To add Annotation items and groups
+window.FusionCharts.addAnnotations = (functionName, id, args) => {
+  var annotations = FusionCharts(id).annotations;
+  annotations[functionName].apply(annotations, args);
+}
+
+//Generic Method to call any fusion chart method exluding except above methods//
+window.FusionCharts.invokeChartFunction = (functionName, chartID, ...args) => {
+
+  var currentChart = FusionCharts(chartID);
+
+  if (functionName === 'addEventListener' || (args.length > 0 && args[0][0] === "callback")) {
```

Comment:
please use my 3.22 mac branch to update this code, it has generic code instead for this

Reply:
done.